src/Chapter1.hs

```diff
@@ -542,7 +545,28 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+sortSortedLists :: [Int] -> [Int] -> [Int]
+sortSortedLists xs [] = xs
+sortSortedLists [] ys = ys
+sortSortedLists (x:xs) (y:ys)
+  | x <= y    = x : sortSortedLists xs (y:ys)
+  | otherwise = y : sortSortedLists (x:xs) ys
+
+
+mergeSort :: [Int] -> [Int]
+mergeSort [] = []
+mergeSort (x:[]) = [x]
+mergeSort (x:y:[]) = min x y : max x y : []
+mergeSort xs = do
+  let sortedFirstHalf = mergeSort (fst halves)
+  let sortedSecondHalf = mergeSort (snd halves)
+  sortSortedLists sortedFirstHalf sortedSecondHalf
+    where halves = splitAt (length xs `div` 2) xs
+
+
+mid :: Int -> Int -> Int -> Int
+mid x y z = sorted !! 1
+  where sorted = mergeSort [x, y, z]
```

Comment:
Just one minor addition: you can implement this even simpler using `let in` and pattern-matching directly:

```haskell
mid x y z = let [_, m, _] = mergeSort [x, y, z] in m
```